### PR TITLE
ports to master (part 3)

### PR DESCRIPTION
**Pull request by Dikay900**
#3667 - f39d351
#3946 - b73894b

### CODE REVIEW

File: modules/objdetect/src/hog.cpp [hunk 1]
```diff
@@ -1453,6 +1453,7 @@ void HOGDescriptor::detect(const Mat& img,
     Size winStride, Size padding, const std::vector<Point>& locations) const
 {
     hits.clear();
+    weights.clear();
```
Comment (StevenPuttemans) [inline modules/objdetect/src/hog.cpp]: O I followed up on this fix in 2.4, where it was only 1 line, but like remarked in the PR history, in master there are several `hits.clear()` lines that should be followed by the `weights.clear()` line.

**Comment by Dikay900** [comment]
Yes thanks for the note i didn't find the other line in the first place.
This should be enough i think. Cannot find another "hits.clear()"

**Comment by StevenPuttemans** [comment]
Well if you look at https://github.com/Itseez/opencv/blob/master/modules/objdetect/src/hog.cpp
Then you see that there is a hits.clear in the normal and the ocl implementation. Both need this fix.

**Comment by Dikay900** [comment]
there are currently no weights in the ocl implementation for now.

**Comment by StevenPuttemans** [comment]
Yeah missed that point ^_^

**Comment by Dikay900** [comment]
@StevenPuttemans can you approve this PR?

**Comment by StevenPuttemans** [comment]
Ow if it was only possible I would have done earlier. But I have not been granted merging or approving rights!

**Comment by Dikay900** [comment]
ah ok then i bump @vpisarev for this little changes

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**